elastic-apm-agent/src/main/java/co/elastic/apm/agent/premain/ShadedClassLoader.java

```diff
@@ -80,23 +101,39 @@ public ShadedClassLoader(File jar, ClassLoader parent, String customPrefix) thro
 
     private Class<?> defineClass(String name, byte[] classBytes) {
         String packageName = getPackageName(name);
-        if (packageName != null && getPackage(packageName) == null) {
+        if (packageName != null && isPackageNotDefined(packageName)) {
             try {
                 if (manifest != null) {
-                    definePackage(name, manifest, jarUrl);
+                    definePackage(packageName, manifest, jarUrl);
                 } else {
                     definePackage(packageName, null, null, null, null, null, null, null);
                 }
-            } catch (IllegalArgumentException e) {
+            } catch (IllegalArgumentException e){
                 // The package may have been defined by a parent class loader in the meantime
-                if (getPackage(packageName) == null) {
+                if (isPackageNotDefined(packageName)) {
                     throw e;
                 }
             }
         }
         return defineClass(name, classBytes, 0, classBytes.length, ShadedClassLoader.class.getProtectionDomain());
     }
 
+    @SuppressWarnings("deprecation")
+    private boolean isPackageNotDefined(String packageName) {
```

Comment:
Probably a matter of taste but I'd always use affirmative rather than negative method names.
See also https://softwareengineering.stackexchange.com/a/196837